src/query.ts

```diff
@@ -226,6 +273,78 @@ export class Query {
     this.setFilters(filters, 'exclude');
   }
 
+  private _findSortField(field: string, sortParams: Array<RequestSortOptions>): number {
+    let index = -1;
+    sortParams.forEach((sort: RequestSortOptions, idx: number) => {
+      if (field in (sort as GenericObject)) {
+        index = idx;
+      }
+    });
+
+    return index;
+  }
+
+  /**
+   * Remove a sorting field from the parameters
+   *
+   * @param   {String}    field   The field we want to remove the ordering
+   *
+   */
+  public removeSorting(field: string): void {
+    if (!this.params.sort) {
+      return;
+    }
+
+    const sortParams = this.params.sort;
+    const index = this._findSortField(field, sortParams as Array<RequestSortOptions>);
+    if (index !== -1) {
+      (sortParams as Array<RequestSortOptions>).splice(index, 1);
+      if (sortParams.length === 0) {
+        delete this.params.sort;
+      } else {
+        this.params.sort = sortParams;
+      }
+    }
+  }
+
+  /**
+   * Adds or alters a sorting parameter to the query
+   *
+   * @param  {String}   field   The field to order by
+   *
+   * @param  {String}   order   Order literal to order by
+   *
+   */
+  public addSorting(field: string, order = Sort.ASC): void {
+    // This is so Prettier won't change it to const automagically
+    let sortParams: RequestSortOptions = [];
+    sortParams = this.params.sort || [];
+    this.removeSorting(field);
+    const obj: GenericObject = {};
+    obj[field] = order;
+    (sortParams as Array<RequestSortOptions>).push(obj);
+
+    this.params.sort = sortParams;
+  }
+
+  /**
+   * Sets all the sorting options at once
+   *
+   * @param  {Array}  sortings    The sortings in the order wanted
+   *
+   */
+  public setSorting(sortings: Array<RequestSortOptions>): void {
```

Comment:
I would just have sort() to set sortings (`sort('field')`, `sort({field: order})`, `sort({field: order}, {field2: order2}, …)`, `sort(null)`, or just `sort()` to reset).

To get sortings I would have `getSort()`.

I think it's the easiest API (`sort()` may be confusing but I think it's not too bad…).

Reply:
I'm leaving this as is, because it matches the Filters API and thus, it will be easier to remember; so we have add*, set* and remove* and they will hide the gritty details from the user of the object, except for set* which allows a more fine-grained operation.

Reply:
I think it's a bad idea to manage sorting array in a way other than as a whole.